modules/util/source-formatter/src/main/java/com/liferay/source/formatter/check/XMLLog4jLoggersCheck.java

```diff
@@ -93,24 +96,29 @@ private void _checkLoggers(String fileName, String content)
 		}
 	}
 
-	private Set<String> _getSrcPaths() throws Exception {
-		if (!_srcPaths.isEmpty()) {
+	private synchronized List<String> _getSrcPaths() throws Exception {
+		if (_srcPaths != null) {
 			return _srcPaths;
 		}
 
+		_srcPaths = new ArrayList<>();
+
 		File file = getPortalDir();
 
 		List<String> fileNames = getFileNames(
 			file.getAbsolutePath(), new String[0],
 			new String[] {"**/com/liferay/**/*.java"});
 
 		for (String fileName : fileNames) {
+			fileName = StringUtil.replace(
+				fileName, CharPool.BACK_SLASH, CharPool.SLASH);
+
 			_srcPaths.add(fileName.substring(fileName.indexOf("com/liferay/")));
 		}
 
 		return _srcPaths;
 	}
 
-	private final Set<String> _srcPaths = new HashSet<>();
+	private List<String> _srcPaths;
```

Comment:
@qz1010 We should still use a `Set`. There's a case where the current `srcPath` can be found in `srcPaths` immediately and we won't have to iterate through `srcPaths` (constant-time performance). So please keep it as as `HashSet`. Otherwise, we will always be doing a linear search.